File: dhandho-core/src/main/java/cc/dhandho/jsclient/ng/CommandResponse.java
package cc.dhandho.jsclient.ng;

public class CommandResponse extends JsonWrapper {
	
	public String command;
	
	public Object json;

	public long responseTime;
	
	public long requestTime;
	
	public CommandResponse(long requestTime, String command, Object json) {
		this.requestTime = requestTime;
		this.json = json;
		this.command = command;
		responseTime = System.currentTimeMillis();
		
	}
	
	public long getTimeCost() {
		return this.responseTime - this.requestTime;
	}

	

}

File: dhandho-jsclient/src/app/command/command-response.ts
export class CommandResponse {
    public command: string;
    public json: any;
    public responseTime: number;
    public requestTime: number;
    public constructor( requestTime: number, command: string, json: any ) {

        if ( this.command === undefined ) {
            this.command = null;
        }
        if ( this.json === undefined ) {
            this.json = null;
        }
        if ( this.responseTime === undefined ) {
            this.responseTime = 0;
        }
        if ( this.requestTime === undefined ) {
            this.requestTime = 0;
        }
        this.requestTime = requestTime;
        this.json = json;
        this.command = command;
        this.responseTime = /* currentTimeMillis */Date.now();
    }

    public getTimeCose(): number {
        return this.responseTime - this.requestTime;
    }
}

